### PR TITLE
fix(parser): html entities evaluated (#704)

### LINKED ISSUE #704: 🐛 BUG: HTML entities are being evaluated, e.g. &copy; -> © instead of keeping them as-is
### What package manager are you using?

npm

### What operating system are you using?

Windows

### Describe the Bug

HTML entities are being evaluated, e.g. `&copy;` -> © instead of keeping them as-is.

### Steps to Reproduce

1. Clone the minimal reproducible example repository
2. `npm install`
3. `npm run build`
4. Check the contents of `dist/index.html`. The `<body>` tag should contain `©` character instead of the original `&copy;`
5. Expected to retain `&copy;` as-is.

### Link to Minimal Reproducible Example (Optional)

https://github.com/Janne252/-snowpackjs-astro-issue-704-HTML-entities-are-being-evaluated

### PR DESCRIPTION
**Pull request by sno2**
Closes #704

## Changes

- This change fixes the behavior of evaluating html entities.

## Testing

I added tests using Cheerio.  However, Cheerio evaluates some of the entities as well so the tests.

## Docs

bug fix only

### CODE REVIEW

File: packages/astro/test/fixtures/html-encoded-characters/src/pages/index.astro [hunk 1]
```diff
@@ -0,0 +1,11 @@
+---
+---
+<html>
+<head><title>HTML Encoded Characters</title></head>
+<body>
+  <h1>&nbsp;&nbsp;&nbsp;Hello, world&semi;</h1>
+	<div>
+		<p>Nested elements&quest; No problem&period;&nbsp;</p>
```
Comment (FredKSchott) [inline packages/astro/test/fixtures/html-encoded-characters/src/pages/index.astro:8]: It looks like the parser handles these differently, so can you add a test for inside of an HTML attribute?

```
<p data-test="&quest;">
```

**Review by FredKSchott — COMMENTED** [review]
LGTM with one comment

**Comment by FredKSchott** [comment]
Merging now to fix our docs site, but would love for that test to still be added in a followup!

**Comment by jasikpark** [comment]
image
:0 doesn't look like it worked out

**Comment by sno2** [comment]
@FredKSchott @jasikpark This was fixing #704, not #727.  [This comment](https://github.com/snowpackjs/astro/issues/727#issuecomment-882633956) explains what is causing the issue with markdown.

**Comment by jasikpark** [comment]
Ah, so there's different systems escaping and unescaping random HTML entities I guess?

**Comment by sno2** [comment]
> Ah, so there are different systems escaping and unescaping random HTML entities I guess?

Yes, it seems so.

**Comment by FredKSchott** [comment]
@sno2 whoops, look like this didn't fix things. had to revert: fcfc625

You can use this page of the docs site to test your fix: `/core-concepts/astro-components#comparing-astro-versus-jsx`

Example: this from was your branch with the PR running on it: https://astro-docs-git-fork-sno2-fix-html-entities-evaluated-pikapkg.vercel.app/core-concepts/astro-components#comparing-astro-versus-jsx

**Comment by sno2** [comment]
Yeah, working on it now.  It seems we did some flaky serialize/deserialize with certain html entities and I broke it.  Furthermore, cheerio actually [decodes some html entities](https://github.com/cheeriojs/cheerio/issues/49) on its own which makes it even worse to test.

**Pull request merged**